[Obs AI Assistant] Do not attempt to overwrite index mappings if inference id in component template has changed

sorenlouv opened the pull request:
Improve flaky API tests and avoid errors like:

```
[ERROR][plugins.observabilityAIAssistant.service] Failed to PUT mapping for .kibana-observability-ai-assistant-kb: illegal_argument_exception
 	Root causes:
 		illegal_argument_exception: Mapper for [semantic_text] conflicts with existing mapper:
 	Cannot update parameter [inference_id] from [pt_tiny_elser_inference_id] to [pt_tiny_text_embedding_inference_id]
```


<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

Commit:
[Obs AI Assistant] Do not override index mappings with changed inference id

Commit:
Retry inference creation in test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300319)
* Commit: 10f1e01950cc73af949331dd795cce0d66cfc15e
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-220739-10f1e01950cc`

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300319","number":300319,"commit":"10f1e01950cc73af949331dd795cce0d66cfc15e"}],"number":300319}
buildkite-pr-comment-->

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8261](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8261)
[✅] x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.ai_assistant.serverless.config.ts: 100/100 tests passed.
[❌] x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts: 75/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F220739%2Fhead)

Commit:
Catch “ the model id is the same as the deployment id of a current model deployment” error

 the model id is the same as the deployment id of a current model deployment]: Cannot create model [pt_tiny_text_embedding] the model id is the same as the deployment id of a current model deployment","attributes":{"body":{"error":{"root_cause":[{"type":"status_exception","reason":"Cannot create model [pt_tiny_text_embedding] the model id is the same as the deployment id of a current model deployment"}],"type":"status_exception","reason":"Cannot create model [pt_tiny_text_embedding] the model id is the same as the deployment id of a current model deployment"},"status":400}}}'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300322)
* Commit: b036aab1f9bfd9642e1a626ef70e0e85788df7b7
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-220739-b036aab1f9bf`

### History
* :green_heart: [Build #300319](https://buildkite.com/elastic/kibana-pull-request/builds/300319) succeeded 10f1e01950cc73af949331dd795cce0d66cfc15e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300322","number":300322,"commit":"b036aab1f9bfd9642e1a626ef70e0e85788df7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300319","number":300319,"commit":"10f1e01950cc73af949331dd795cce0d66cfc15e"}],"number":300322}
buildkite-pr-comment-->

Commit:
Add retry logic to “Knowledge base: when changing from ELSER to E5-like model” test

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300325)
* Commit: b48ceb57aa76a87117dabb77b39ec23b205491b3

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/300325#0196c078-7d2b-4d11-aa6f-90087c9920ac)

### History
* :green_heart: [Build #300322](https://buildkite.com/elastic/kibana-pull-request/builds/300322) succeeded b036aab1f9bfd9642e1a626ef70e0e85788df7b7
* :green_heart: [Build #300319](https://buildkite.com/elastic/kibana-pull-request/builds/300319) succeeded 10f1e01950cc73af949331dd795cce0d66cfc15e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300325","number":300325,"commit":"b48ceb57aa76a87117dabb77b39ec23b205491b3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300322","number":300322,"commit":"b036aab1f9bfd9642e1a626ef70e0e85788df7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300319","number":300319,"commit":"10f1e01950cc73af949331dd795cce0d66cfc15e"}],"number":300325}
buildkite-pr-comment-->

Commit:
Remove unused ml dep

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300326)
* Commit: 6d3375037a0074f6c8b0e4769b105798a7501313
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-220739-6d3375037a00`

### History
* :broken_heart: [Build #300325](https://buildkite.com/elastic/kibana-pull-request/builds/300325) failed b48ceb57aa76a87117dabb77b39ec23b205491b3
* :green_heart: [Build #300322](https://buildkite.com/elastic/kibana-pull-request/builds/300322) succeeded b036aab1f9bfd9642e1a626ef70e0e85788df7b7
* :green_heart: [Build #300319](https://buildkite.com/elastic/kibana-pull-request/builds/300319) succeeded 10f1e01950cc73af949331dd795cce0d66cfc15e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300326","number":300326,"commit":"6d3375037a0074f6c8b0e4769b105798a7501313"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300325","number":300325,"commit":"b48ceb57aa76a87117dabb77b39ec23b205491b3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300322","number":300322,"commit":"b036aab1f9bfd9642e1a626ef70e0e85788df7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300319","number":300319,"commit":"10f1e01950cc73af949331dd795cce0d66cfc15e"}],"number":300326}
buildkite-pr-comment-->

Commit:
Merge branch 'main' into do-not-overwrite-index-mappings-with-different-inference-id

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300330)
* Commit: 44fc61d43dd31f6dddaafa4988eac4dc12205266
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-220739-44fc61d43dd3`

### History
* :green_heart: [Build #300326](https://buildkite.com/elastic/kibana-pull-request/builds/300326) succeeded 6d3375037a0074f6c8b0e4769b105798a7501313
* :broken_heart: [Build #300325](https://buildkite.com/elastic/kibana-pull-request/builds/300325) failed b48ceb57aa76a87117dabb77b39ec23b205491b3
* :green_heart: [Build #300322](https://buildkite.com/elastic/kibana-pull-request/builds/300322) succeeded b036aab1f9bfd9642e1a626ef70e0e85788df7b7
* :green_heart: [Build #300319](https://buildkite.com/elastic/kibana-pull-request/builds/300319) succeeded 10f1e01950cc73af949331dd795cce0d66cfc15e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300330","number":300330,"commit":"44fc61d43dd31f6dddaafa4988eac4dc12205266"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300326","number":300326,"commit":"6d3375037a0074f6c8b0e4769b105798a7501313"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300325","number":300325,"commit":"b48ceb57aa76a87117dabb77b39ec23b205491b3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300322","number":300322,"commit":"b036aab1f9bfd9642e1a626ef70e0e85788df7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300319","number":300319,"commit":"10f1e01950cc73af949331dd795cce0d66cfc15e"}],"number":300330}
buildkite-pr-comment-->

Commit:
Fix “has an index created in 8.18” test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300337)
* Commit: 12dff964950b509c7400d60e6d449eef2e9529fb
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-220739-12dff964950b`

### History
* :green_heart: [Build #300330](https://buildkite.com/elastic/kibana-pull-request/builds/300330) succeeded 44fc61d43dd31f6dddaafa4988eac4dc12205266
* :green_heart: [Build #300326](https://buildkite.com/elastic/kibana-pull-request/builds/300326) succeeded 6d3375037a0074f6c8b0e4769b105798a7501313
* :broken_heart: [Build #300325](https://buildkite.com/elastic/kibana-pull-request/builds/300325) failed b48ceb57aa76a87117dabb77b39ec23b205491b3
* :green_heart: [Build #300322](https://buildkite.com/elastic/kibana-pull-request/builds/300322) succeeded b036aab1f9bfd9642e1a626ef70e0e85788df7b7
* :green_heart: [Build #300319](https://buildkite.com/elastic/kibana-pull-request/builds/300319) succeeded 10f1e01950cc73af949331dd795cce0d66cfc15e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300337","number":300337,"commit":"12dff964950b509c7400d60e6d449eef2e9529fb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300330","number":300330,"commit":"44fc61d43dd31f6dddaafa4988eac4dc12205266"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300326","number":300326,"commit":"6d3375037a0074f6c8b0e4769b105798a7501313"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300325","number":300325,"commit":"b48ceb57aa76a87117dabb77b39ec23b205491b3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300322","number":300322,"commit":"b036aab1f9bfd9642e1a626ef70e0e85788df7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300319","number":300319,"commit":"10f1e01950cc73af949331dd795cce0d66cfc15e"}],"number":300337}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/14975898963

sorenlouv commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/220791">](https://github.com/elastic/kibana/pull/220791)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @sorenlouv

sorenlouv commented:
Backported to 8.19 in https://github.com/elastic/kibana/pull/220791
